GameEngineTests/GameBoardTests.cs

```diff
@@ -25,41 +25,112 @@ public void ShouldStartOwlsAtFirstSpots()
         [DataRow(CardType.Purple, 5)]
         public void ShouldMakeSimpleMoves(CardType cardType, int newPosition)
         {
-            var board = new GameBoard(2);
+            var board = new GameBoard(2, 1);
+            var play = new Play(cardType, 0);
 
-            board.Move(cardType);
+            board.Move(play);
 
-            Assert.AreEqual(newPosition, board.OwlPosition);
+            Assert.AreEqual(newPosition, board.OwlPositions[0]);
         }
 
         [TestMethod]
         public void ShouldMoveOwlIntoNest()
         {
-            var board = new GameBoard(2);
-            board.OwlPosition = board.Board.Count - 2;
+            var board = new GameBoard(2, 1);
+            var initialPosition = board.Board.Count - 2;
+            board.OwlPositions[0] = initialPosition;
+            var play = new Play(CardType.Red, initialPosition);
 
-            board.Move(CardType.Red);
+            board.Move(play);
 
-            Assert.AreEqual(board.Board.Count - 1, board.OwlPosition);
-            Assert.AreEqual(BoardPositionType.Nest, board.Board[board.OwlPosition]);
+            var newOwlPosition = board.OwlPositions[0];
+            Assert.AreEqual(board.NestPosition, newOwlPosition,
+                "Owl is not on last position.");
+            Assert.AreEqual(BoardPositionType.Nest, board.Board[newOwlPosition],
+                "Owl's position is not a nest.");
         }
 
         [TestMethod]
-        public void ShouldFailToMoveOwlWhenItIsAlreadyInTheNest()
+        public void ShouldFailToMoveWhenThereAreNoOwlsAtThePosition()
```

Comment:
❓ The intent behind `ShouldFailToMoveOwlWhenItIsAlreadyInTheNest` was that you cannot make a `Play` which targets an Owl already in the Nest. I don't think that is captured in this test.

Could you add a new test for this case?

Reply:
Restored a version of `ShouldFailToMoveOwlWhenItIsAlreadyInTheNest`.